### PR TITLE
[IP-385] healtcheck function commons to all azure function implementation

### PR DESCRIPTION
**Pull request by gquadrati**
#### List of Changes

Added base healtcheck functions, common to all Azure Functions projects.

Usage:

```` typescript

    pipe(
      getConfig(),
//     ^^^ the env config (with `azureStorageUrl` and `anUrlToCheck`)
      checkApplicationHealth(IConfig, 
//                            ^^^ the io-ts Codec
     [
        c => checkAzureStorageHealth(c.azureStorageUrl),
        c => checkUrlHealth(c.anUrlToCheck)
      ]
//       ^^^ a list of checks, built with decoded config and one of the base check functions
      
      ),
...
);

````

#### Motivation and Context

Remove duplicated code

#### How Has This Been Tested?

unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by michaeldisaro — APPROVED** [review]
LGTM. Apart some naming fixes in tests.

File: src/utils/__tests__/healthcheck.test.ts [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    expect.assertions(1);
+
+    pipe(
+      checkAzureCosmosDbHealth("", ""),
+      TE.map(_ => {
+        expect(true).toBeTruthy();
+        done();
+      }),
+      TE.mapLeft(_ => {
+        console.log(_);
```
[the remark this answers is not in the document]
Comment (balanza) [reply src/utils/__tests__/healthcheck.test.ts:214]: jm2c: I usually name a variable `_` when:
* I don't use it
* I don't care what's in it, I just handle it like a whole black box

I guess here we are in the latter scenario

[the remark this answers is not in the document]
Comment (balanza) [reply src/utils/__tests__/healthcheck.test.ts:214]: Anyway: I'd remove logs from tests

File: src/utils/healthcheck.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+/**
+ * Check application's configuration is correct
+ *
+ * @returns either true or an array of error messages
+ */
+export const checkConfigHealth: <IConfig>(
+  ConfigCodec: t.Type<IConfig>
+) => (value: unknown) => HealthCheck<"Config", IConfig> = <IConfig>(
+  ConfigCodec: t.Type<IConfig>
+) => (value: unknown): HealthCheck<"Config", IConfig> =>
```
Comment (balanza) [inline src/utils/healthcheck.ts:53]: We may want to have cleaner signature:
```suggestion
export const checkConfigHealth = <IConfig>(
  ConfigCodec: t.Type<IConfig>
) => (value: unknown): HealthCheck<"Config", IConfig> =>
```

File: src/utils/__tests__/healthcheck.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (gquadrati) [reply src/utils/__tests__/healthcheck.test.ts:214]: Yes, actually I can even remove all of them, in this tests logs are not necessary.
Anyway, I use `_` like @balanza said

[the remark this answers is not in the document]
Comment (michaeldisaro) [reply src/utils/__tests__/healthcheck.test.ts:214]: Ok, for "I don't care about it". Seems legit.

File: src/utils/healthcheck.ts (hunk 2 above)
Comment (gquadrati) [reply src/utils/healthcheck.ts:53]: Sure, thanks

File: src/utils/__tests__/healthcheck.test.ts [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    expect.assertions(1);
+
+    pipe(
+      checkUrlHealth(""),
+      TE.map(_ => {
+        expect(false).toBeTruthy();
+        done();
+      }),
+      TE.mapLeft(_ => {
+        console.log(_);
```
[the remark this answers is not in the document]
Comment (gquadrati) [reply src/utils/__tests__/healthcheck.test.ts]: Removed [here](https://github.com/pagopa/io-functions-commons/pull/203/commits/97627d26d6fa441768d5a6a90216344ef7bd1df9)

File: src/utils/__tests__/healthcheck.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (gquadrati) [reply src/utils/__tests__/healthcheck.test.ts]: Removed [here](https://github.com/pagopa/io-functions-commons/pull/203/commits/97627d26d6fa441768d5a6a90216344ef7bd1df9)

**Comment by gquadrati** [comment]
@balanza @michaeldisaro can we merge it?

**Comment by michaeldisaro** [comment]
> @balanza @michaeldisaro can we merge it?

Ok for me. These functionalities are not in conflict with any system and we can test their integration.

**Pull request merged**